debug.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func testDebugDir(path string) error {
+	info, err := os.Stat(path)
+	if err != nil {
+		err := os.MkdirAll(path, 0755)
+		return err
+	}
+
+	if !info.IsDir() {
+		return errors.New("Debugdir exists, but is a file not a directory")
+	}
+
+	return nil
+}
+
+func copyPackage(pkgName, src, dest string) error {
+	infos, err := ioutil.ReadDir(src)
+	if err != nil {
+		return err
+	}
+
+	destPkg := filepath.Join(dest, pkgName)
```

Comment:
remember that for a package `foo/bar`, we want the output to be at `${dir}/foo/bar/z0.go`, not `${dir}/bar/z0.go`.

otherwise, if we have two packages `foo/bar` and `other/bar`, both sets of files will get mixed up.

Reply:
Yes that's exactly what it does already.

Reply:
gotcha. this should be clearer then. we should do `filepath.Join(dest, "foo", "bar")` instead of `filepath.Join(dest, "foo/bar")`.This might work on Windows, but it's still counter-intuitive. How about `components := strings.Split(pkgPath, "/")`?